packages/next/src/shared/lib/constants.ts

```diff
@@ -59,7 +59,13 @@ export const DEV_MIDDLEWARE_MANIFEST = '_devMiddlewareManifest.json'
 export const REACT_LOADABLE_MANIFEST = 'react-loadable-manifest.json'
 export const AUTOMATIC_FONT_OPTIMIZATION_MANIFEST = 'font-manifest.json'
 export const SERVER_DIRECTORY = 'server'
-export const CONFIG_FILES = ['next.config.js', 'next.config.mjs']
+export const CONFIG_FILES = [
+  'next.config.js',
+  'next.config.mjs',
+  'next.config.ts',
+  'next.config.cts',
```

Comment:
I'm thinking of dropping `.cts` and `.mts` support since:

```sh
Default CJS Project

// .ts -> takes ESM format -> .js
// .cts -> takes ESM format -> .js (WHY? Just use .ts)
// .mts -> takes ESM format -> .js (WHY? Just use .ts)

ESM Project (package.json type: module)

// .ts -> takes ESM format -> .cjs as CommonJS
// .cts -> takes ESM format -> .cjs as CommonJS (WHY? Just use .ts)
// .mts -> takes ESM format -> .cjs as CommonJS (WHY? Just use .ts)
```